src/lib.rs

```diff
@@ -81,45 +104,82 @@ pub struct Gdbm {
 impl Gdbm {
     // API: open database file, read and validate header
     pub fn open(pathname: &str, dbcfg: &GdbmOptions) -> io::Result<Gdbm> {
-        // derive open options
-        let opt_write: bool = !dbcfg.readonly;
-        let opt_create = if dbcfg.readonly { false } else { dbcfg.creat };
+        if dbcfg.readonly && (dbcfg.newdb || dbcfg.creat) {
+            Err(Error::new(
+                ErrorKind::Other,
+                "readonly conflicts with newdb or creat",
+            ))?;
+        }
 
-        // open filesystem file
         let mut f = OpenOptions::new()
             .read(true)
-            .write(opt_write)
-            .create(opt_create)
+            .write(!dbcfg.readonly)
+            .create(dbcfg.creat | dbcfg.newdb)
             .open(pathname)?;
-        let metadata = f.metadata()?;
 
-        // read gdbm global header
-        let header = Header::from_reader(&dbcfg.alignment, &metadata, &mut f)?;
-        println!("{:?}", header);
+        if dbcfg.newdb {
+            f.set_len(0)?;
+        }
```

Comment:
This is the purpose of `truncate` in `OpenOptions`

Reply:
:+1: